include/arkode/arkode_sprkstep.h

```diff
@@ -0,0 +1,127 @@
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * -----------------------------------------------------------------
+ * This is the header file for the ARKode SPRKStep module.
+ * -----------------------------------------------------------------*/
+
+#ifndef _SPRKSTEP_H
+#define _SPRKSTEP_H
+
+#include <arkode/arkode.h>
+#include <arkode/arkode_sprk.h>
+#include <sundials/sundials_nvector.h>
+
+#ifdef __cplusplus /* wrapper to enable C++ usage */
+extern "C" {
+#endif
+
+/* -----------------
+ * SPRKStep Constants
+ * ----------------- */
+
+static const int SPRKSTEP_DEFAULT_1  = ARKODE_SYMPLECTIC_EULER_1_1;
+static const int SPRKSTEP_DEFAULT_2  = ARKODE_SYMPLECTIC_LEAPFROG_2_2;
+static const int SPRKSTEP_DEFAULT_3  = ARKODE_SYMPLECTIC_MCLACHLAN_3_3;
+static const int SPRKSTEP_DEFAULT_4  = ARKODE_SYMPLECTIC_MCLACHLAN_4_4;
+static const int SPRKSTEP_DEFAULT_5  = ARKODE_SYMPLECTIC_MCLACHLAN_5_6;
+static const int SPRKSTEP_DEFAULT_6  = ARKODE_SYMPLECTIC_YOSHIDA_6_8;
+static const int SPRKSTEP_DEFAULT_8  = ARKODE_SYMPLECTIC_MCLACHLAN_8_16;
+static const int SPRKSTEP_DEFAULT_10 = ARKODE_SYMPLECTIC_SOFRONIOU_10_36;
+
+/* -------------------
+ * Exported Functions
+ * ------------------- */
+
+/* Create, Resize, and Reinitialization functions */
+SUNDIALS_EXPORT void* SPRKStepCreate(ARKRhsFn f1, ARKRhsFn f2, realtype t0,
+                                     N_Vector y0, SUNContext sunctx);
+
+SUNDIALS_EXPORT int SPRKStepReInit(void* arkode_mem, ARKRhsFn f1, ARKRhsFn f2,
+                                   realtype t0, N_Vector y0);
+
+SUNDIALS_EXPORT int SPRKStepReset(void* arkode_mem, realtype tR, N_Vector yR);
+
+/* Rootfinding functions */
+
+/* Rootfinding initialization */
+SUNDIALS_EXPORT int SPRKStepRootInit(void* arkode_mem, int nrtfn, ARKRootFn g);
+
+/* Optional input functions -- must be called AFTER SPRKStepCreate */
+SUNDIALS_EXPORT int SPRKStepSetDefaults(void* arkode_mem);
+SUNDIALS_EXPORT int SPRKStepSetUseCompensatedSums(void* arkode_mem,
+                                                  sunbooleantype onoff);
+SUNDIALS_EXPORT int SPRKStepSetMethod(void* arkode_mem,
+                                      ARKodeSPRKStorage sprk_storage);
+SUNDIALS_EXPORT int SPRKStepSetMethodName(void* arkode_mem, const char* method);
+SUNDIALS_EXPORT int SPRKStepSetOrder(void* arkode_mem, int maxord);
+SUNDIALS_EXPORT int SPRKStepSetInterpolantType(void* arkode_mem, int itype);
+SUNDIALS_EXPORT int SPRKStepSetInterpolantDegree(void* arkode_mem, int degree);
+/* TODO(CJB): should we remove this from the initial release and wait for the OO
+ * adaptivity? */
+SUNDIALS_EXPORT int SPRKStepSetAdaptivityFn(void* arkode_mem, ARKAdaptFn hfun,
+                                            void* h_data);
+SUNDIALS_EXPORT int SPRKStepSetMaxNumSteps(void* arkode_mem, long int mxsteps);
+SUNDIALS_EXPORT int SPRKStepSetMaxHnilWarns(void* arkode_mem, int mxhnil);
```

Comment:
I recommend that this be removed from the public UI until time adaptivity is enabled.